core/chains/evm/client/simulated_backend_client.go

```diff
@@ -1,8 +1,18 @@
+/*
+The simulated backend cannot access old blocks and will return an error if
+anything other than `latest`, `nil`, or the latest block are passed to
+`CallContract`.
+
+The simulated client avoids the old block error from the simulated backend by
+passing `nil` to `CallContract` when calling `CallContext` or `BatchCallContext`
+and will not return an error when an old block is used.
+*/
```

Comment:
Does godoc pick this up as a package level comment? Is there a particular type we could associate it with instead?

Reply:
I added it to the top so that the behavior of two functions is well explained and visible. I could move the explanation to each of the functions in question `CallContext` and `BatchCallContext` as well to ensure the explanation is visible to IDE tools as well.

Reply:
There isn't a `doc.go` file, but I can create one and move this comment there too.

Reply:
Hmmm I thought a `doc.go` was just a convention for humans to find it in source. I don't think godoc and pkg.go.dev care about the file boundaries though :thinking: 
I usually run `godoc -http=:6060` locally to try it out

Reply:
Yeah it just puts all the detached floating file comments like this at the top in the overview:
![image](https://github.com/smartcontractkit/chainlink/assets/1194128/25360735-a32a-4a2e-ad39-7d2ab729cfa3)

Reply:
Yeah, unless there is a specific identifier to associate it with instead.